posts/2024-05-03-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,417 @@
+
+=== Jakarta Data
+
+The https://jakarta.ee/specifications/data/[Jakarta Data] specification, part of https://jakarta.ee/[Jakarta EE], proposes a unified API for simplified data access across different types of databases, from relational to NoSQL databases.
+
+Jakarta Data achieves its goal by introducing concepts like Repositories and custom query methods, improving the developer experience when using data retrieval and manipulation APIs.
+
+[NOTE]
+Jakarta Data is planned for officially becoming part of https://jakarta.ee/specifications/data/1.0/[Jakarta EE 11]
+
+=== Eclipse JNoSQL: A Reference Implementation
+
+A Jakarta EE Specification can't solve the problem by itself - it becomes consumable through an _implementation_.
+
+Each Jakarta EE Specification has at least one implementation. The existance of an implementation proves the proposed specification is achievable and can be developed by interested third-parties. This is when companies and communities actively starts providing their own implementations, empowering the Jakarta EE developers and ecosystem with a diverse and powerful toolset.
+
+[NOTE]
+    Example of reference implementations (RI) are Hibernate for Jakarta Persistence 3.1 specification; Jersey for Jakarta RESTFul Web Services 3.1 specification; Glassfish for Jakarta Servlet 6.0 specification; Weld for Jakarta Context And Dependency Injection (CDI) 4.0 specification; and so on...
+
+https://projects.eclipse.org/projects/technology.jnosql[*Eclipse JNoSQL*] is a compatible implementation of the *Jakarta NoSQL* and *Jakarta Data* specifications, a framework that streamlines the integration of Java applications with relational and NoSQL databases. Powered by the Jakarta Contexts and Dependency Injection (CDI) specification, it is compatible with Jakarta EE and Eclipse MicroProfile compatible solutions.
+
+Eclipse JNoSQL covers four NoSQL database types: *key-value*, *column*, *document* and *graph* databases.
+
+[NOTE]
+Currently, the Jakarta NoSQL doesn't define an API for Graph database types but Eclipse JNoSQL provides a Graph template to explore the specific behavior of this database type by using https://tinkerpop.apache.org/[Apache TinkerPop] as a communication layer.
+
+As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+==== Why Eclipse JNoSQL?
+
+In the code samples below, note the APIs similarities and differences accross different vendors. All samples demonstrate a commonly used functionality for Document databases: a document creation and the addition of a property to that document:
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/mongodb.png[MongoDB ,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new Document();
+document.append(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/ArangoDB.png[Arango DB,align="left" width=25%,height=25%]
```

Comment:
none of the DB icons are rendering and I suspect our adoc implementation doesn't support URLs as image values. You could either remove the image or add the pngs to the PR

Reply:
I added the pngs to the PR!